### PR TITLE
tests: add analytic kernel herding tests.

### PR DESCRIPTION
**Pull request by pc532627**
Refs: #786
Closes: #786 

### PR Type

- Documentation content changes
- Tests

### Description

Kernel herding now has an analytical test (for the case where unique is set to true and when it is set to false) and these tests have been used to showcase to a user a minimal worked example in the documentation.

### How Has This Been Tested?

Two new tests added and pass.

### Does this PR introduce a breaking change?

N/A 

### Checklist before requesting a review
- [x] I have made sure that my PR is not a duplicate.
- [x] My code follows the style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have performed a self-review of my code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: tests/unit/test_solvers.py [hunk 1]
```diff
@@ -524,6 +524,315 @@ def test_herding_state(self, reduce_problem: _ReduceProblem) -> None:
         expected_state = HerdingState(solver.kernel.gramian_row_mean(dataset))
         assert eqx.tree_equal(state, expected_state)
 
+    def test_kernel_herding_analytic_unique(self) -> None:
+        # pylint: disable=line-too-long
+        r"""
+        Test kernel herding on an analytical example, enforcing a unique coreset.
+
+        In this example, we have data of:
```
Comment (qh681248) [inline tests/unit/test_solvers.py:532]: Firstly, this is such a good example and explanation and makes KernelHerding so easy to understand. Thank you for adding it :)
`In this example, we have data of:` can become
`In this example, we have a data of:`

File: tests/unit/test_solvers.py [hunk 2]
```diff
@@ -524,6 +524,315 @@ def test_herding_state(self, reduce_problem: _ReduceProblem) -> None:
...
+                0.5 & 0.125
+            \end{pmatrix}
+
+        and choose a ``length_scale`` of :math:`\frac{1}{\sqrt{2}}` to simplify
+        computations with the ``SquaredExponentialKernel``, in particular it becomes:
+
+        .. math::
+            k(x, y) = e^{-||x - y||^2}
+
+        in this example.
```
Comment (qh681248) [inline tests/unit/test_solvers.py:547]: `in this example` can be removed.

File: tests/unit/test_solvers.py [hunk 3]
```diff
@@ -524,6 +524,315 @@ def test_herding_state(self, reduce_problem: _ReduceProblem) -> None:
...
+        np.testing.assert_array_equal(
+            coreset.unweighted_indices, expected_coreset_indices
+        )
+        np.testing.assert_array_equal(
+            coreset.coreset.data, data.data[expected_coreset_indices]
+        )
+        np.testing.assert_array_almost_equal(
+            solver_state.gramian_row_mean, expected_gramian_row_mean
+        )
+
```
Comment (qh681248) [inline tests/unit/test_solvers.py:836]: This is excellent! SonarQube might complain that some of the code is repeated/can be refactored but I am a fan of clarity.

**Review by qh681248 — APPROVED** [review]

File: tests/unit/test_solvers.py (hunk 1 above)
Comment (pc532627) [reply tests/unit/test_solvers.py:532]: Thanks - I think using data without the a preceding it is what we do in the rest of the codebase but happy to be changed if we don't

File: tests/unit/test_solvers.py (hunk 3 above)
Comment (pc532627) [reply tests/unit/test_solvers.py:991]: Added a weighted example

File: tests/unit/test_solvers.py [hunk 4]
```diff
@@ -833,6 +829,165 @@ def test_kernel_herding_analytic_not_unique(self) -> None:
...
+                0.1
+            \end{pmatrix}
+
+        and choose a ``length_scale`` of :math:`\frac{1}{\sqrt{2}}` to simplify
+        computations with the ``SquaredExponentialKernel``, in particular it becomes:
+
+        .. math::
+            k(x, y) = e^{-||x - y||^2}.
+
+        # TODO: Update from here
```
Comment (qh681248) [inline tests/unit/test_solvers.py:861]: Thank you so much for adding an example for weighted data as well, it is a very clear explanation. I believe you forgot to remove `# TODO`, Apart from that, I am happy to merge it :)

**Review by qh681248 — APPROVED** [review]

**Pull request merged**